tagstudio/src/qt/widgets/video_player.py

```diff
@@ -0,0 +1,382 @@
+        self.play_pause.hide()
+        self.mute_button.hide()
+        # self.fullscreen_button.hide()
+        return super().releaseMouse()
+
+    def resetControlsToDefault(self) -> None:
+        # Resets the video controls to their default state.
+        self.play_pause.load("./tagstudio/resources/pause.svg")
+        self.mute_button.load("./tagstudio/resources/volume_muted.svg")
+
+    def pauseToggle(self) -> None:
+        if self.player.isPlaying():
+            self.player.pause()
+            self.play_pause.load("./tagstudio/resources/play.svg")
+        else:
+            self.player.play()
+            self.play_pause.load("./tagstudio/resources/pause.svg")
+
+    def muteToggle(self) -> None:
+        if self.player.audioOutput().isMuted():
+            self.player.audioOutput().setMuted(False)
+            self.mute_button.load("./tagstudio/resources/volume_unmuted.svg")
+        else:
+            self.player.audioOutput().setMuted(True)
+            self.mute_button.load("./tagstudio/resources/volume_muted.svg")
+
+    def play(self, filepath: str, resolution: QSize) -> None:
+        # Sets the filepath and sends the current player position to the very end, so that the new video can be played.
+        # self.player.audioOutput().setMuted(True)
+        logging.info(f"Playing {filepath}")
+        self.resolution = resolution
+        self.filepath = filepath
+        if self.player.isPlaying():
+            self.player.setPosition(self.player.duration())
+            self.player.play()
+        else:
+            self.checkMediaStatus(QMediaPlayer.MediaStatus.EndOfMedia)
+        # logging.info(f"Successfully stopped.")
+
+    def stop(self) -> None:
+        self.filepath = None
+        self.player.stop()
+
+    def resizeVideo(self, new_size: QSize) -> None:
+        # Resizes the video preview to the new size.
+        self.video_preview.setSize(new_size)
+        self.video_tint.setRect(
+            0, 0, self.video_preview.size().width(), self.video_preview.size().height()
+        )
+
+        rcontent = self.contentsRect()
+        self.centerOn(self.video_preview)
+        self.roundCorners()
+        self.setSceneRect(0, 0, rcontent.width(), rcontent.height())
+        self.keepControlsInPlace()
+
+    def roundCorners(self) -> None:
+        mask = Image.new(
+            "RGBA",
+            (
+                int(self.contentsRect().size().width()),
+                int(self.contentsRect().size().height()),
+            ),
+            (0, 0, 0, 255),
+        )
+        draw = ImageDraw.Draw(mask)
+        draw.rounded_rectangle(
+            (0, 0)
+            + (self.contentsRect().size().width(), self.contentsRect().size().height()),
+            radius=20,
+            fill=(0, 0, 0, 0),
+        )
+        mask.resize(
+            (
+                int(self.contentsRect().size().width()),
+                int(self.contentsRect().size().height()),
+            )
+        )
```

Comment:
Recommend ensuring that the width and height of the video are always at least 0 to prevent a `ValueError: Width and height must be >= 0` error which prevents any new video from loading (I actually ran into this). The changes to this method also include removing the unnecessary resize step as well as decreasing the border radius to better match the image thumbnails.
```suggestion
        width: int = int(max(self.contentsRect().size().width(), 0))
        height: int = int(max(self.contentsRect().size().height(), 0))
        mask = Image.new(
            "RGBA",
            (
                width,
                height,
            ),
            (0, 0, 0, 255),
        )
        draw = ImageDraw.Draw(mask)
        draw.rounded_rectangle(
            (0, 0) + (width, height),
            radius=12,
            fill=(0, 0, 0, 0),
        )
```